apps/web/src/actions/datasets/generateDataset.ts

```diff
@@ -0,0 +1,91 @@
+'use server'
+
+import {
+  ChainObjectResponse,
+  Dataset,
+  StreamEventTypes,
+} from '@latitude-data/core/browser'
+import { createDataset } from '@latitude-data/core/services/datasets/create'
+import { ChainEventDto, Latitude } from '@latitude-data/sdk'
+import env from '$/env'
+import { getCurrentUser } from '$/services/auth/getCurrentUser'
+import { createStreamableValue } from 'ai/rsc'
+
+type GenerateDatasetActionProps = {
+  parameters: Record<string, unknown>
+  description: string
+  rowCount: number
+  name: string
+}
+
+export async function generateDatasetAction({
+  parameters,
+  description,
+  rowCount,
+  name,
+}: GenerateDatasetActionProps) {
+  let response: Dataset | undefined
+  const { user, workspace } = await getCurrentUser()
+  const stream = createStreamableValue<
+    { event: StreamEventTypes; data: ChainEventDto },
+    Error
+  >()
+  const gateway = {
+    host: env.GATEWAY_HOSTNAME,
+    port: env.GATEWAY_PORT,
+    ssl: env.GATEWAY_SSL,
+  }
+  const sdk = new Latitude('d7c94cc1-72a8-4252-b1a8-3e6b3cf9c29f', {
+    projectId: 20,
+    gateway,
+  })
```

Comment:
Hardcoded doc uuid? This will definitely fail on the opensource

Reply:
lol 💯

Reply:
Also, what happens in the local env? Does it send the request to prod, or does it search for a local prompt with that projectId and docUuid?

Reply:
the later